action.php

```diff
@@ -72,24 +73,41 @@ public function convert(Doku_Event $event) {
         // it's ours, no one else's
         $event->preventDefault();
 
-        // prepare cache and its dependencies
-        $depends = array();
-        $cache = $this->prepareCache($depends);
+        $generateNewPdf = false;
+        if ($REV || $DATE_AT) {
+            $tempFilename = tempnam('/tmp', 'dw2pdf_');
+            $generateNewPdf = true;
+        } else {
+            // prepare cache and its dependencies
+            $depends = array();
+            $cache = $this->prepareCache($depends);
+            $tempFilename = $cache->cache;
+        }
 
         // hard work only when no cache available or needed for debugging
-        if(!$this->getConf('usecache') || $this->getExportConfig('isDebug') || !$cache->useCache($depends)) {
+        $generateNewPdf = $generateNewPdf
+                       || !$this->getConf('usecache')
+                       || $this->getExportConfig('isDebug')
+                       || !$cache->useCache($depends);
+        if($generateNewPdf) {
             // generating the pdf may take a long time for larger wikis / namespaces with many pages
             set_time_limit(0);
+            try {
+                $this->generatePDF($tempFilename);
+            } catch (Mpdf\MpdfException $e) {
+                //prevent act_export()
+                $ACT = 'show';
+                msg($e->getMessage(), -1);
+                return false;
+            }
 
-            $this->generatePDF($cache->cache);
         }
 
         // deliver the file
-        $this->sendPDFFile($cache->cache);
+        $this->sendPDFFile($tempFilename);
```

Comment:
We should probably delete the temporary file again?

Reply:
implemented.